packages/squiggle-lang/__tests__/TS/KDE.ts

```diff
@@ -0,0 +1,59 @@
+import * as fc from "fast-check";
+import { samplesToContinuousPdf } from "../../src/rescript/Distributions/SampleSetDist/KdeLibrary.js";
+import range from "lodash/range";
+import sumBy from "lodash/sumBy";
+
+// KDE by definition, with triangular kernel max(0, |1-x|)
+const evalKde = (samples: number[], width: number, weight: number, x: number) =>
+  (weight / width) *
+  sumBy(samples, (p) => Math.max(0, 1 - Math.abs((x - p) / width)));
+
+describe("Kernel density estimation", () => {
+  test("should approximately equal naive definition", () => {
+    fc.assert(
+      fc.property(
+        fc.array(fc.double({ min: -10, max: 10, noNaN: true }), {
+          minLength: 6,
+        }),
+        fc.integer({ min: 4, max: 1e3 }),
+        fc.double({ min: 0.1, max: 3, noNaN: true }),
+        (samples, size, wantedWidth) => {
+          const len = samples.length;
+          const weight = 1 / len;
+          const sortedSamples = samples.sort((a, b) => a - b);
+          if (sortedSamples[len - 1] - sortedSamples[0] < 2 * wantedWidth)
+            return true;
+
+          const {
+            xs,
+            ys,
+            usedWidth: width,
+          } = samplesToContinuousPdf(sortedSamples, size, wantedWidth, weight);
+
+          expect(xs.length).toEqual(size);
+          expect(ys.length).toEqual(size);
```

Comment:
Is it not possible to put each of these blocks in its own ``test()`` function?

Reply:
This is a lot of different ``expect`` calls for one ``test`` call. I imagine we could either split up the ``test`` call, or put the ``test`` call under the ``property`` call.